stac/src/main/scala/StacMediaType.scala

```diff
@@ -0,0 +1,56 @@
+package geotrellis.server.stac
+
+import cats.implicits._
+import io.circe._
+
+sealed abstract class StacMediaType(val repr: String) {
+  override def toString: String = repr
+}
+
+object StacMediaType {
+
+  private def fromString(s: String): StacMediaType = s match {
+    case "image/tiff" => `image/tiff`
+    case "image/vnd.stac.geotiff" => `image/vnd.stac.geotiff`
+    case "image/vnd.stac.geotiff; cloud-optimized=true" => `image/cog`
```

Comment:
I'm hesitant about it as well -- but the spec specifically lists a bunch of common media type strings https://github.com/radiantearth/stac-spec/blob/master/item-spec/item-spec.md#media-types, so I'm hopeful that spec implementers will copy-paste instead of manually typing unpleasant strings

Reply:
where did `image/cog` come from? I've never seen that used before and I don't think it's a valid one, as everything in `image` is supposed to be in the RFC.

Reply:
`image/cog` exists only in Scala -- for STAC implementers, the media type will be the string in the case statement. It's just a shorthand that gets serialized to / deserialized from the correct value.

Reply:
It was my guess (and executive decision I suppose) that someone who knows they have a COG and is writing Scala code won't think "I have a `image/vnd.stac.geotiff; cloud-optimized=true`" and that ergonomics would be slightly better if the proper media type happened with only `image/cog`